splink/internals/term_frequencies.py

```diff
@@ -79,33 +80,48 @@ def _join_tf_to_df_concat_sql(linker: Linker) -> str:
     return sql
 
 
-def _join_new_table_to_df_concat_with_tf_sql(linker: Linker, new_tablename: str) -> str:
+def _join_new_table_to_df_concat_with_tf_sql(
+    linker: Linker,
+    input_tablename: str,
+    input_table: Optional[SplinkDataFrame] = None,
+) -> str:
     """
-    Joins any required tf columns onto new_tablename
+    Joins any required tf columns onto input_tablename
 
     This is needed e.g. when using linker.compare_two_records
     or linker.inference.find_matches_to_new_records in which the user provides
     new records which need tf adjustments computed
     """
+    tf_cols_already_populated = []
```

Comment:
The user can now include tf columns in the input data.  If they do, do not recompute, and use whatever the user has provided